app/scripts/offscreen.js

```diff
@@ -0,0 +1,42 @@
+import { OffscreenCommunicationTarget } from '../../shared/constants/offscreen-communication';
+
+/**
+ * Creates an offscreen document that can be used to load additional scripts
+ * and iframes that can communicate with the extension through the chrome
+ * runtime API. Only one offscreen document may exist, so any iframes required
+ * by extension can be embedded in the offscreen.html file. See the offscreen
+ * folder for more details.
+ */
+export async function createOffscreen() {
+  const { chrome } = globalThis;
+  if (!chrome.offscreen || (await chrome.offscreen.hasDocument())) {
+    return;
+  }
+
+  const loadPromise = new Promise((resolve) => {
+    chrome.runtime.onMessage.addListener((msg) => {
```

Comment:
We should remove this listener once we're done with it. We get _a lot_ of messages, this would get called frequently.

Reply:
Done!